File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/reactiveobject/ReadHistoryDataRequest.java
package org.opcfoundation.webserver.service.message.reactiveobject;

import org.eclipse.milo.opcua.stack.core.types.UaStructuredType;
import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;
import org.opcfoundation.webserver.addressspace.nodes.UaVariableType;
import org.opcfoundation.webserver.types.common.UaChildId;
import org.opcfoundation.webserver.types.common.UaObjectId;
import org.springframework.lang.Nullable;

public class ReadHistoryDataRequest {
    private final UaObjectId objectId;
    private final UaChildId childId;
    private UaStructuredType details;
    private final int offset;

    public ReadHistoryDataRequest(
            UaObjectId objectId,
            UaChildId childId,
            UaStructuredType details,
            int offset)
    {
        this.objectId = objectId;
        this.childId = childId;
        this.details = details;
        this.offset = offset;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public UaChildId getChildId() {
        return childId;
    }

    public UaStructuredType getDetails() {
        return details;
    }

    public int getOffset() {
        return offset;
    }
}

File: opcua-webservice-node/src/server/service/message/reactiveobject/ReadHistoryDataRequest.ts
import { UaExtensionObject } from "opcua-webapi-ts";
import { UaChildId, UaObjectId } from "../../../types";

export class ReadHistoryDataRequest {
    private readonly _objectId: UaObjectId;
    private readonly _childId: UaChildId;
    private readonly _details: UaExtensionObject;
    private readonly _offset: number;

    constructor(
        objectId: UaObjectId,
        childId: UaChildId,
        details: UaExtensionObject,
        offset: number,
    ) {
        this._objectId = objectId;
        this._childId = childId;
        this._details = details;
        this._offset = offset;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get childId(): UaChildId {
        return this._childId;
    }

    get details(): UaExtensionObject {
        return this._details;
    }

    get offset(): number {
        return this._offset;
    }
}